src/extractor/graph_compressor.cpp

```diff
@@ -245,8 +249,17 @@ void GraphCompressor::Compress(const std::unordered_set<NodeID> &barrier_nodes,
                                                    roads_on_the_right,
                                                    roads_on_the_left);
                     scripting_environment.ProcessTurn(extraction_turn);
-                    node_duration_penalty = extraction_turn.duration * 10;
-                    node_weight_penalty = extraction_turn.weight * weight_multiplier;
+
+                    if (has_forward_signal)
```

Comment:
Nit: Let’s consider extracting small function to avoid code duplication here. Also for me 10 feels as magic number here - may be extract to constant?